podman/domain/manifests.py

```diff
@@ -0,0 +1,247 @@
+"""Model and Manager for Manifest resources."""
+import urllib.parse
+from typing import List, Optional, Union
+
+import requests
+
+from podman import api
+from podman.api import APIClient
+from podman.domain.images import Image
+from podman.domain.manager import Manager, PodmanResource
+from podman.errors.exceptions import APIError, ImageNotFound, NotFound
+
+
+class Manifest(PodmanResource):
+    """Details and configuration for a manifest managed by the Podman service."""
+
+    @property
+    def id(self) -> str:
+        """Returns the identifier of the manifest."""
+        try:
+            return self.attrs["manifests"][0]["digest"]
+        except (KeyError, TypeError):
+            return ""
+
+    @property
+    def name(self) -> str:
+        """Returns the identifier of the manifest."""
+        try:
+            return self.names[0]
+        except TypeError:
+            raise ValueError("Manifest 'names' missing or empty.")
```

Comment:
The "empty" case is not checked by the code, only the "missing" case.

Reply:
Good catch, I'll also improve the english.